#! /usr/bin/env python

# Declare parameters that control haar-based people detection

PACKAGE='detection'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

##############################################
## multiple_objects_detection:object detector parameters ##
##############################################


gen.add("HMax", int_t, 0, "Hue max", 180, 0, 360)
gen.add("SMax", int_t, 0, "Saturation max", 256, 0, 255)
gen.add("VMax", int_t, 0, "Value max", 256, 0, 255)
gen.add("HMin", int_t, 0, "Hue min", 0, 0, 360)
gen.add("SMin", int_t, 0, "Saturation min", 0, 0, 255)
gen.add("VMin", int_t, 0, "Value min", 100, 0, 255)

gen.add("DMin", int_t, 0, "Depth min", 0, 0, 255)
gen.add("DMax", int_t, 0, "Depth max", 255, 0, 255)

gen.add("h_bins", int_t, 0, "Divide (HMin,HMax) to h_bins parts", 18, 1, 100)
gen.add("s_bins", int_t, 0, "Divide (SMin,SMax) to s_bins parts", 16, 1, 100)
gen.add("v_bins", int_t, 0, "Divide (VMin,VMax) to s_bins parts", 8, 1, 100)
gen.add("d_bins", int_t, 0, "Divide (DMin,DMax) to d_bins parts", 255, 1, 255)

gen.add("AREA_TOLERANCE", int_t, 0, "AREA_TOLERANCE", 20, 5, 100)
gen.add("QUALITY_TOLERANCE", int_t, 0, "QUALITY_TOLERANCE", 6000, 100, 65535)
gen.add("DENSITY_TOLORENCE", double_t, 0, "DENSITY_TOLORENCE", 0.2, 0, 50)



exit(gen.generate(PACKAGE, "detection", "multiple_objects_detection"))
